src/pipecat_flows/manager.py

```diff
@@ -167,25 +110,41 @@ def __init__(
         self.current_functions: Set[str] = set()  # Track registered functions
         self.current_node: Optional[str] = None
 
-    async def initialize(self, initial_messages: List[dict]) -> None:
-        """Initialize the flow with starting messages.
+    async def initialize(self, initial_messages: Optional[List[dict]] = None) -> None:
+        """Initialize the flow.
 
-        For static flows, also sets the initial node from config.
+        For static flows, sets the initial system message from config.
+        For dynamic flows, uses provided initial_messages.
 
         Args:
-            initial_messages: Initial system messages for the LLM
+            initial_messages: Optional initial system messages for dynamic flows.
+                            Ignored for static flows.
 
         Raises:
-            FlowInitializationError: If initialization fails
+            FlowInitializationError: If initialization fails or if initial_messages
+                                   is missing for dynamic flows.
         """
         if self.initialized:
             logger.warning(f"{self.__class__.__name__} already initialized")
             return
 
         try:
-            # Set initial context with no tools
-            await self.task.queue_frame(LLMMessagesUpdateFrame(messages=initial_messages))
-            await self.task.queue_frame(LLMSetToolsFrame(tools=[]))
+            if self.nodes:  # Static flow
+                if initial_messages:
+                    logger.warning("Initial messages ignored for static flow configuration")
+
+                # Set initial system message if present
+                if hasattr(self, "initial_system_message"):
```

Comment:
I think this could be:

```
if self.initial_system_message:
```

We should always have things (i.e. fields) defined, because this is kind of hiding what fields are defined or not depending on something else.

Reply:
Makes sense!